test/e2e/framework/network/utils.go

```diff
@@ -817,15 +817,15 @@ func (config *NetworkingTestConfig) setup(selector map[string]string) {
 		family = v1.IPv6Protocol
 		secondaryFamily = v1.IPv4Protocol
 	}
-	// Get Node IPs from the cluster, ExternalIPs take precedence
-	config.NodeIP = e2enode.FirstAddressByTypeAndFamily(nodeList, v1.NodeExternalIP, family)
```

Comment:
Make this parametrizable, I've added functional options some time ago, you can do something lile
```
// PreferExternalAddresses prefer node External Addresses for the tests
func PreferExternalAddresses(config *NetworkingTestConfig) {
	config.PreferExternalAddresses = true
}
```
```

Reply:
and here you just use the bool to pick the address

```

if config.PreferExternalAddress {
  config.NodeIP = e2enode.FirstAddressByTypeAndFamily(nodeList, v1. NodeExternalIP, family)
}
if config.NodeIP == "" {
		config.NodeIP = e2enode.FirstAddressByTypeAndFamily(nodeList, v1.NodeInternalIP, family)
}
```

or something like this